src/lazy/system_reader.rs

```diff
@@ -378,19 +378,35 @@ impl<Encoding: Decoder, Input: IonInput> SystemReader<Encoding, Input> {
                 "expected a macro table definition operation, but found: {operation:?}"
             ));
         }
-        let mut macro_table = MacroTable::new();
+        let mut macro_table = MacroTable::empty();
```

Comment:
🗺️ When reading a `(macro_table ...)` clause, the reader no longer pre-populates the macro table with system macros.